src/lib/profiles/software-update/SoftwareUpdateProfile.cpp

```diff
@@ -28,460 +28,575 @@
 #include <Weave/Profiles/ProfileCommon.h>
 #include "SoftwareUpdateProfile.h"
 using namespace ::nl::Weave;
-using namespace ::nl::Weave::Profiles;
-using namespace ::nl::Weave::Profiles::SoftwareUpdate;
-/*
- * -- definitions for ImageQuery and its supporting classes --
- *
- * the no-arg constructor for an IntegrityTypeList
+
+namespace nl {
+namespace Weave {
+namespace Profiles {
+namespace SoftwareUpdate {
+
+/**
+ * The default constructor for an IntegrityTypeList.  Constructs a logically empty list.  The list may be populated via the init()
+ * method or by deserializing the list from a message.
  */
 IntegrityTypeList::IntegrityTypeList()
 {
-  theLength=0;
-  for (int i=0; i<3; i++) theList[i]=kIntegrityType_SHA160;
+    theLength = 0;
+    for (int i = 0; i < 3; i++)
+        theList[i] = kIntegrityType_SHA160;
 }
-/*
- * parameters:
- * - uint8_t aLength, an 8-bit length value for the list (<4)
- * - uint8_t *aList, pointer to an array of itegrity type values
- * return:
- * - WEAVE_NO_ERROR if it's all good
- * - WEAVE_ERROR_INVALID_LIST_LENGTH if the length is too long
+
+/**
+ * Explicitly initialize the IntegrityTypeList with an list of supported IntegrityTypes
+ *
+ * @param[in] aLength   An 8-bit value for the length of the list.  Must be less that the number of enums in @ref IntegrityTypes.
+ * @param[in] aList  A pointer to an array of @ref IntegrityTypes values.  May be NULL only if aLength is 0.
+ *
+ * @retval WEAVE_NO_ERROR  On success.
+ * @retval WEAVE_ERROR_INVALID_LIST_LENGTH If the length is too long
  */
-WEAVE_ERROR IntegrityTypeList::init(uint8_t aLength, uint8_t *aList)
+WEAVE_ERROR IntegrityTypeList::init(uint8_t aLength, uint8_t * aList)
 {
-  if (aLength>3) return WEAVE_ERROR_INVALID_LIST_LENGTH;
-  theLength=aLength;
-  for (int i=0; i<theLength; i++) theList[i]=aList[i];
-  return WEAVE_NO_ERROR;
+    if (aLength > 3)
```

Comment:
Worth making this an assertion?

Reply:
I kept the return error code in the init functions.